code/modules/mechs/components/_components.dm

```diff
@@ -67,23 +69,33 @@
 /obj/item/mech_component/proc/ready_to_install()
 	return TRUE
 
+
 /obj/item/mech_component/proc/repair_brute_damage(amt)
 	take_brute_damage(-amt)
+	if(total_damage <= 0)
+		return
 
 /obj/item/mech_component/proc/repair_burn_damage(amt)
 	take_burn_damage(-amt)
+	if(total_damage <= 0)
+		return
+
 
 /obj/item/mech_component/proc/take_brute_damage(amt)
 	brute_damage += amt
 	update_health()
 	if(total_damage == max_damage)
 		take_component_damage(amt,0)
+	if(amt < 1)
+		return
```

Comment:
```suggestion
	if(amt < 1)
		return
	if(total_damage == max_damage)
		take_component_damage(amt,0)
```

Reply:
Please review, I changed the code to something else =)